app/src/common/shared/org/mozilla/vrbrowser/browser/SettingsStore.java

```diff
@@ -116,10 +117,20 @@ public void setTelemetryEnabled(boolean isEnabled) {
         final boolean hasEnabled = isTelemetryEnabled();
         if (hasEnabled != isEnabled) {
             TelemetryWrapper.init(mContext);
+
+            if (!GleanMetricsService.isInitialized()) {
```

Comment:
Did you try to see how it works without this lines? I still think that's unnecessary.

Reply:
you are right.

Reply:
Mh, I still see these lines of code though ... did you push the latest updates?